homeassistant/components/radio_browser/media_source.py

```diff
@@ -0,0 +1,158 @@
+
+from .const import DOMAIN
+
+CODEC_TO_MIMETYPE = {
+    "MP3": "audio/mpeg",
+    "AAC": "audio/aac",
+    "AAC+": "audio/aac",
+    "OGG": "application/ogg",
+}
+
+
+async def async_get_media_source(hass: HomeAssistant) -> RadioMediaSource:
+    """Set up Radio Browser media source."""
+    # Radio browser support only a single config entry
+    entry = hass.config_entries.async_entries(DOMAIN)[0]
+    radios = hass.data[DOMAIN]
+
+    return RadioMediaSource(hass, radios, entry)
+
+
+@callback
+def _async_get_station_mime_type(station: Station) -> str | None:
+    """Determine mime type of a radio station."""
+    mime_type = CODEC_TO_MIMETYPE.get(station.codec)
+    if not mime_type:
+        mime_type, _ = mimetypes.guess_type(station.url)
+    return mime_type
+
+
+class RadioMediaSource(MediaSource):
+    """Provide Radio stations as media sources."""
+
+    def __init__(
+        self, hass: HomeAssistant, radios: RadioBrowser, entry: ConfigEntry
+    ) -> None:
+        """Initialize CameraMediaSource."""
+        super().__init__(DOMAIN)
+        self.hass = hass
+        self.entry = entry
+        self.radios = radios
+
+    async def async_resolve_media(self, item: MediaSourceItem) -> PlayMedia:
+        """Resolve selected Radio station to a streaming URL."""
+        station = await self.radios.station(uuid=item.identifier)
+        if not station:
+            raise BrowseError("Radio station is no longer available")
+
+        if not (mime_type := _async_get_station_mime_type(station)):
+            raise BrowseError("Could not determine stream type of radio station")
+
+        # Register "click" with Radio Browser
+        await self.radios.station_click(uuid=station.uuid)
+
+        return PlayMedia(station.url, mime_type)
+
+    async def async_browse_media(
+        self,
+        item: MediaSourceItem,
+    ) -> BrowseMediaSource:
+        """Return media."""
+        root = BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=None,
+            media_class=MEDIA_CLASS_CHANNEL,
+            media_content_type=MEDIA_TYPE_MUSIC,
+            title=self.entry.title,
+            can_play=False,
+            can_expand=True,
+            children_media_class=MEDIA_CLASS_DIRECTORY,
+        )
+        root.children = []
+
+        # Browsing stations
+        if item.identifier:
+            stations = []
+            if item.identifier == "popular":
+                stations = await self.radios.stations(
+                    hide_broken=True, limit=256, order=Order.CLICK_COUNT, reverse=True
+                )
+            else:
+                stations = await self.radios.stations(
+                    filter_by=FilterBy.COUNTRY_CODE_EXACT,
+                    filter_term=item.identifier,
+                    hide_broken=True,
+                    order=Order.CLICK_COUNT,
+                    reverse=True,
+                )
+
+            if stations:
```

Comment:
This if-guard is not necessary if stations is an empty list

Reply:
Ah right, the previous library was able to return None, this one handle that.

Can be removed. Thanks!